### PR TITLE
ci: add soak tests with aks clusters

### LINKED ISSUE #170: Add periodic soak tests with AKS clusters
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Azure Key Vault provider version: (use the image tag):
- Kubernetes version: (use `kubectl version`):
- Cluster type: (e.g. AKS, aks-engine, etc):

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes #170 

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

**Comment by aramase** [comment]
Soak test results: https://dev.azure.com/azure/secrets-store-csi-driver-provider-azure/_build/results?buildId=19502&view=results

**Comment by aramase** [comment]
Soak test results: https://dev.azure.com/azure/secrets-store-csi-driver-provider-azure/_build/results?buildId=19502&view=results

File: .pipelines/nightly.yaml [hunk 1]
```diff
@@ -21,3 +21,8 @@ jobs:
       osTypes:
       - "linux"
       - "windows"
+  - template: soak-tests-template.yaml
+    parameters:
+      clusterConfigs:
+      - "csi-secrets-store-soak-linux-aks"
+      - "csi-secrets-store-soak-win-aks"
```
Comment (ritazh) [inline .pipelines/nightly.yaml:28]: where are these configs?

Comment (aramase) [reply .pipelines/nightly.yaml:28]: They are downloaded as part of soak tests: https://github.com/Azure/secrets-store-csi-driver-provider-azure/pull/396/files#diff-22792ae5cdc74eefcbce99909d002257cadcf0328af36b1995eb1c5ac7dbc657R33-R45

**Review by ritazh — APPROVED** [review]

**Pull request merged**